### PR TITLE
Add test config for GCE GPU stable2-stable1 upgrade test.

### PR DESCRIPTION
**Pull request by jiayingz**

### CODE REVIEW

File: testgrid/config.yaml [hunk 1]
```diff
@@ -4027,6 +4027,8 @@ dashboards:
     test_group_name: ci-kubernetes-e2e-gce-stable2-stable1-upgrade-cluster
   - name: gce-1.9-1.10-upgrade-cluster-new
     test_group_name: ci-kubernetes-e2e-gce-stable2-stable1-upgrade-cluster-new
+  - name: gce-1.9-1.10-upgrade-gpu
+    test_group_name: ci-kubernetes-e2e-gce-stable2-stable1-gpu-upgrade
```
[the remark this answers is not in the document]
Comment (jiayingz) [reply testgrid/config.yaml]: Thanks for the info!

File: jobs/config.json [hunk 2]
```diff
@@ -3640,6 +3640,28 @@
       "sig-cli"
     ]
   },
+  "ci-kubernetes-e2e-gce-stable2-stable1-gpu-upgrade": {
+    "args": [
+      "--check-leaked-resources",
+      "--check-version-skew=false",
+      "--env=STORAGE_MEDIA_TYPE=application/vnd.kubernetes.protobuf",
```
Comment (mindprince) [inline jobs/config.json]: Why is this needed?

File: jobs/config.json [hunk 3]
```diff
@@ -3640,6 +3640,28 @@
...
+      "--env-file=jobs/env/ci-kubernetes-e2e-gce-device-plugin-gpu.env",
+      "--env-file=jobs/env/ci-cri-containerd-e2e-gce.env",
```
Comment (mindprince) [inline jobs/config.json]: Why is this needed?

File: prow/config.yaml [hunk 4]
```diff
@@ -10990,6 +10990,19 @@ periodics:
...
+- interval: 12h
+  agent: kubernetes
+  name: ci-kubernetes-e2e-gce-stable2-stable1-gpu-upgrade
+  labels:
+    preset-service-account: "true"
+    preset-k8s-ssh: "true"
+  spec:
+    containers:
+    - args:
+      - --timeout=920
```
Comment (mindprince) [inline prow/config.yaml]: This timeout seems rather long.

File: testgrid/config.yaml [hunk 5]
```diff
@@ -4378,6 +4382,11 @@ dashboards:
     base_options: 'exclude-filter-by-regex=^BeforeSuite$'
     alert_options:
       alert_mail_to_addresses: 'gke-kubernetes-accelerators-bugs@google.com'
+  - name: gce-stable2-stable1-gpu-upgrade
```
Comment (mindprince) [inline testgrid/config.yaml]: Let's call this gce-1.9-1.10-gpu-upgrade

File: jobs/config.json (hunk 2 above)
Comment (jiayingz) [reply jobs/config.json]: For upgrade test, I think one needs to specify STORAGE_MEDIA_TYPE for etcd. I have used STORAGE_MEDIA_TYPE=application/json in my manual runs of upgrade test to make it work, but looks like STORAGE_MEDIA_TYPE=application/vnd.kubernetes.protobuf has been used in the existing ci-kubernetes-e2e-gce-stable2-stable1-upgrade-cluster tests.

File: jobs/config.json (hunk 3 above)
Comment (jiayingz) [reply jobs/config.json]: This was copied from the existing ci-cri-containerd-e2e-gce-device-plugin-gpu, which I guess is not really needed as the test is not focusing on containerd integration. Removed.

File: prow/config.yaml (hunk 4 above)
Comment (jiayingz) [reply prow/config.yaml]: Changed to 180.

File: testgrid/config.yaml (hunk 5 above)
Comment (jiayingz) [reply testgrid/config.yaml]: done.
